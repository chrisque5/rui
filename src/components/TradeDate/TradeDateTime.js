import PropTypes from 'prop-types';
import { TimePicker, Form } from 'antd';

const TradeDateTime = ({
  isTradeTimeEnabled, id,
}) => (
  <Form.Item
    className="item-trade-time"
    name={id}
    rules={[{ required: isTradeTimeEnabled, message: 'Trade time required' }]}
  >

    <TimePicker
      disabled={!isTradeTimeEnabled}
      style={{ width: '100%' }}
      placeholder="Trade Time"
      onChange={(time) => time.local()}
    />
  </Form.Item>
);

TradeDateTime.propTypes = {
  isTradeTimeEnabled: PropTypes.bool.isRequired,
  id: PropTypes.string.isRequired,
};

export default TradeDateTime;
