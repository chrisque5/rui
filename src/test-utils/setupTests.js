// The below part of the code is to fix the issue
// TypeError: window.matchMedia is not a function
Object.defineProperty(window, 'matchMedia', {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});
