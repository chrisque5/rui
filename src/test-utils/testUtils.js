/* eslint-disable import/no-extraneous-dependencies, react/prop-types */
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Form } from 'antd';

const middlewares = [thunk];
const renderWithForm = (
  ui,
  {
    initialState,
    store = configureMockStore(middlewares)(initialState),
    ...renderOptions
  } = {},
) => {
  const wrapper = ({ children }) => <Provider store={store}><Form>{children}</Form></Provider>;

  return render(ui, { wrapper, ...renderOptions });
};

const renderWithProvider = (
  ui,
  {
    initialState,
    store = configureMockStore(middlewares)(initialState),
    ...renderOptions
  } = {},
) => {
  const wrapper = ({ children }) => <Provider store={store}>{children}</Provider>;

  return render(ui, { wrapper, ...renderOptions });
};
// re-export everything
export * from '@testing-library/react';
// override render method
export { renderWithForm, renderWithProvider };
