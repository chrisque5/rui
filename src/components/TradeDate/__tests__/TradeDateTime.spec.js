import { renderWithForm, waitFor } from 'test-utils/testUtils';
import TradeDateTime from '../TradeDateTime';

// move this out to a mocks file to be reused
const mockForm = {};
const tradeDateTimeOnChange = jest.fn();
const defaultProps = {
  form: mockForm,
  id: 'TradeDateTime',
  isTradeTimeEnabled: false,
  initialTime: '01012020',
  tradeDateTimeOnChange,
};

test('<TradeDateTime /> renders snapshot as expected', () => {
  const { container } = renderWithForm(<TradeDateTime {...defaultProps} />);
  expect(container).toMatchSnapshot();
});

test('<TradeDateTime /> with prop.isTradeDateTimeEnabled = false renders the DatePicker disabled', async () => {
  const { getByPlaceholderText } = renderWithForm(<TradeDateTime {...defaultProps} />);
  const datePicker = await waitFor(() => getByPlaceholderText('Trade Time'));
  expect(datePicker.disabled).toBe(true);
});

test('<TradeDateTime /> with prop.isTradeDateTimeEnabled = true renders the DatePicker enabled', async () => {
  const props = { ...defaultProps, isTradeTimeEnabled: true };
  const { getByPlaceholderText } = renderWithForm(<TradeDateTime {...props} />);
  const datePicker = await waitFor(() => getByPlaceholderText('Trade Time'));
  expect(datePicker.disabled).toBe(false);
});
