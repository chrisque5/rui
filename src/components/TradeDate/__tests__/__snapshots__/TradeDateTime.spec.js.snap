// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TradeDateTime /> renders snapshot as expected 1`] = `
<div>
  <form
    class="ant-form ant-form-horizontal"
  >
    <div
      class="ant-row ant-form-item item-trade-time"
    >
      <div
        class="ant-col ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <div
              class="ant-picker ant-picker-disabled"
              style="width: 100%;"
            >
              <div
                class="ant-picker-input"
              >
                <input
                  autocomplete="off"
                  disabled=""
                  id="TradeDateTime"
                  placeholder="Trade Time"
                  readonly=""
                  size="10"
                  title=""
                  value=""
                />
                <span
                  class="ant-picker-suffix"
                >
                  <span
                    aria-label="clock-circle"
                    class="anticon anticon-clock-circle"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="clock-circle"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                      />
                      <path
                        d="M686.7 638.6L544.1 535.5V288c0-4.4-3.6-8-8-8H488c-4.4 0-8 3.6-8 8v275.4c0 2.6 1.2 5 3.3 6.5l165.4 120.6c3.6 2.6 8.6 1.8 11.2-1.7l28.6-39c2.6-3.7 1.8-8.7-1.8-11.2z"
                      />
                    </svg>
                  </span>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;
